build/WFLY-19130_publish_Wildfly_channel_manifest.adoc

```diff
@@ -0,0 +1,195 @@
+feature packs has to have a separate channel associated with it.
+
+A channel consists of a stream of manifests with the current manifest being determined by its Maven version. When a new version
+of WildFly is published a new manifest with the same version as Wildfly release will be added to the channel.
+The backwards compatibility of a channel is achieved through a naming convention where the supported version range is added to the 
+artifactId of the manifest (e.g. `org.wildfly:wildfly-32.0`).
+
+For this proposal we will publish the latest manifest with no backwards compatibility guarantee, meaning each channel will always
+contain components of the latest released version of WildFly.
+
+Following three channels will be created as part of this proposal:
+
+[cols=3]
+|===
+| Manifest GA
+| Artifacts
+| Depends on
+
+|`org.wildfly.channels:wildfly-ee`
+|wildfly-ee-galleon-pack and its dependencies
+| -
+
+|`org.wildfly.channels:wildfly`
+| wildfly-galleon-pack and its dependencies
+|`org.wildfly.channels:wildfly-ee`
+
+|`org.wildfly.channels:wildfly-preview`
+| wildfly-preview-galleon-pack and its dependencies
+| -
+
+|===
+
+==== Manifest metadata Requirements
+
+The manifest file defines a few metadata fields:
+* name - human readable name of the manifest. Should contain the version of the server.
+* id - identifier of the manifest used to resolve manifest dependencies. Should be the same as the GA of the manifest.
+* description - human readable description of the manifest.
+* requires - id and optionally maven coordinates of dependent manifests.
+
+=== Hard Requirements
+
+* WildFly manifests are published in Maven repository during release.
+* Provisioning tools using wildfly-channels can use the published manifest to provision and update WildFly server.
+* Different WildFly features packs created in the project are able to use different versions of the same artifacts.
+* The versions of the manifests correspond to Wildfly releases. When using version `32.0.0.Final`, the user should 
+receive manifest for Wildfly 32.0.0 Final
+
+=== Nice-to-Have Requirements
+
+
+=== Non-Requirements
+* The WildFly channels do not guarantee backwards compatibility.
+* Using the WildFly channels to perform a major version upgrade is possible, but not supported.
+* Publishing a channel definition (channel.yaml file)
+
+=== Future Work
+N/A
+
+== Backwards Compatibility
+
+Publishing the manifests adds new capability for provisioning and testing tools, but doesn't affect existing features.
+
+=== Default Configuration
+
+N/A
+
+=== Importing Existing Configuration
+
+N/A
+
+=== Deployments
+
+N/A
+
+=== Interoperability
+N/A
+
+== Security Considerations
+
+N/A
+
+== Test Plan
+
+* Existing WildFly tests already use the channels
+* Add test in Prospero to provision the latest WildFly version
+
+== Community Documentation
+
+Update Prospero documentation and examples with provisioning WildFly server using published manifests.
```

Comment:
We should update https://docs.wildfly.org/wildfly-maven-plugin/releases/4.2/provision-example.html as well.